### PR TITLE
update MQTT docs to include discover_ip option

### PR DESCRIPTION
**Pull request by Links2004**
## Description:

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#6673

## Checklist:

  - [x] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: components/mqtt.rst [hunk 1]
```diff
@@ -137,10 +139,93 @@ Configuration options:
...
+
+MQTT device discovery
+---------------------
+
+The ESPHome device will respond to the following MQTT topics if `mqtt.discover_ip` is enabled.
+
+- `esphome/discover` (All ESPHome device will answer)
+- `esphome/ping/<APP_NAME>`
+
+The response will be send to `esphome/discover/<APP_NAME>` and is a JSON encoded message.
```
Comment (kbx81) [inline components/mqtt.rst]:
```suggestion
The response will be sent to `esphome/discover/<APP_NAME>` and is a JSON encoded message.
```

File: components/mqtt.rst [hunk 2]
```diff
@@ -137,10 +139,93 @@ Configuration options:
...
+
+The MQTT device discovery is currently used for:
+
+- ESPHome dashboard (online / offline status)
+- ESPHome CLI (IP discovery e.g. for upload, logs)
```
Comment (kbx81) [inline components/mqtt.rst]:
```suggestion
- ESPHome CLI (IP discovery; used to view logs and perform OTA uploads)
```

File: components/mqtt.rst [hunk 3]
```diff
@@ -137,10 +139,93 @@ Configuration options:
...
+      "board": "esp32-c3-devkitm-1",
+      "network": "wifi",
+      "api_encryption": "Noise_NNpsk0_25519_ChaChaPoly_SHA256"
+    }
+
+
+JSON keys:
+
+-  **ip** (**Required**, ip): The IP address of the ESPHome device.
+-  **name** (**Required**, string): Name of the device (`esphome.name`).
```
Comment (kbx81) [inline components/mqtt.rst]:
```suggestion
-  **name** (**Required**, string): Name of the device (for example: `my-device.local`).
```

File: components/mqtt.rst [hunk 4]
```diff
@@ -137,10 +139,93 @@ Configuration options:
...
+-  **mac** (**Required**, string): MAC address of the device.
+-  **board** (**Required**, string): Board used for the device.
+-  **version** (**Required**, string): ESPHome version.
+-  **port** (*Optional*, port): Port of the ESPHome API (if activated).
```
Comment (kbx81) [inline components/mqtt.rst]:
```suggestion
-  **port** (*Optional*, port): Port of the ESPHome API (if enabled).
```

File: components/mqtt.rst [hunk 5]
```diff
@@ -137,10 +139,93 @@ Configuration options:
...
+-  **ipX** (*Optional*, ip): Additional IP addresses (X is a number starting at 1).
+-  **friendly_name** (*Optional*, string): Friendly name of the device (`esphome.friendly_name`).
```
Comment (kbx81) [inline components/mqtt.rst]:
```suggestion
-  **friendly_name** (*Optional*, string): Friendly name of the device (for example: `Bedroom lamp`).
```

**Review by kbx81 — DISMISSED** [review]
Just a few minor nitpicks (mostly), but otherwise lgtm

File: components/mqtt.rst (hunk 3 above)
Comment (Links2004) [reply components/mqtt.rst]: `esphome.name` is the name of the key in the yaml of the device, 
which is not coresponding to the mDNS name, so no `.local` will be added.

File: components/mqtt.rst (hunk 5 above)
Comment (Links2004) [reply components/mqtt.rst]: I provided example data in the JSON part, the idea with the `esphome.friendly_name` is to show from where the data is comming in the yaml definition of the device.

File: components/mqtt.rst [hunk 6]
```diff
@@ -137,10 +139,93 @@ Configuration options:
...
-.. _mqtt-using_with_home_assistant:
 
-Using with Home Assistant
--------------------------
+.. _mqtt-device_discovery:
+
+MQTT device discovery
+---------------------
+
+The ESPHome device will respond to the following MQTT topics if `mqtt.discover_ip` is enabled.
```
Comment (jesserockz) [inline components/mqtt.rst]: All inline code should be double backticks, single just makes it italic.
```suggestion
The ESPHome device will respond to the following MQTT topics if ``mqtt.discover_ip`` is enabled.
```

**Review by jesserockz — CHANGES_REQUESTED** [review]

File: components/mqtt.rst (hunk 6 above)
Comment (Links2004) [reply components/mqtt.rst]: changed

**Review by jesserockz — APPROVED** [review]

**Pull request merged**